android-lib/src/main/java/com/kinvey/android/store/DataStore.java

```diff
@@ -682,12 +683,24 @@ public void delete(@NonNull Query query, @NonNull KinveyDeleteCallback callback)
      *
      * @param callback KinveyPushCallback
      */
-    public void push(@NonNull KinveyPushCallback callback){
+    public void push(@NonNull KinveyPushCallback callback) {
         Preconditions.checkNotNull(client, "client must not be null");
         Preconditions.checkArgument(client.isInitialize(), "client must be initialized.");
+        if (KINVEY_API_VERSION_5.equals(kinveyApiVersion)) {
```

Comment:
better check if `kinveyApiVersion` is `greater or equal than` instead of `equal`